testonly/hammer/hammer.go

```diff
@@ -481,6 +480,10 @@ func (s *hammerState) retryOneOp(ctx context.Context) (err error) {
 		}
 
 		if time.Now().After(deadline) {
+			if firstErr == nil {
+				// If there was no other error, we've probably hit the deadline - make sure we bubble that up.
+				firstErr = ctx.Err()
```

Comment:
It's a bit annoying that the context error doesn't have a `DEADLINE_EXCEEDED`  status. Not sure it's worth returning one explicitly though as it won't be handled differently.

Reply:
Yeah, tempted to leave it for now.